### PR TITLE
Add support for separate DB config

### PR DESCRIPTION
**Pull request by tobybatch**

### CODE REVIEW

File: startup.sh [hunk 1]
```diff
@@ -4,13 +4,24 @@ echo $KIMAI
...
+    DB_PORT=$(awk -F '[/:@]' '{print $7}' <<< $DATABASE_URL)
+    DB_BASE=$(awk -F '[/?]' '{print $4}' <<< $DATABASE_URL)
+  else
+    DB_TYPE=${DB_TYPE:mysql}
+    if [ "$DB_TYPE" == "mysql" ]; then
+      export DATABASE_URL="${DB_TYPE}://${DB_USER:=kimai}:${DB_PASS:=kimai}@${DB_HOST:=sqldb}:${DB_PORT:=3306}/${DB_BASE:=kimai}"
+    elif [ "$DB_TYPE" == "sqlite" ]; then
+      export DATABASE_URL="${DB_TYPE}://${DB_BASE:=/%kernel.project_dir%/var/data/kimai.sqlite}"
+    else
+      echo "Unkown dtabase type, cannot proceed. [$DB_TYPE]"
```
Comment (kevinpapst) [inline startup.sh]:
```suggestion
      echo "Unknown database type, cannot proceed. Expected one of: mysql, sqlite. Received: [$DB_TYPE]"
```

File: startup.sh [hunk 2]
```diff
@@ -4,13 +4,24 @@ echo $KIMAI
...
+    DB_PASS=$(awk -F '[/:@]' '{print $5}' <<< $DATABASE_URL)
+    DB_HOST=$(awk -F '[/:@]' '{print $6}' <<< $DATABASE_URL)
+    DB_PORT=$(awk -F '[/:@]' '{print $7}' <<< $DATABASE_URL)
+    DB_BASE=$(awk -F '[/?]' '{print $4}' <<< $DATABASE_URL)
+  else
+    DB_TYPE=${DB_TYPE:mysql}
+    if [ "$DB_TYPE" == "mysql" ]; then
+      export DATABASE_URL="${DB_TYPE}://${DB_USER:=kimai}:${DB_PASS:=kimai}@${DB_HOST:=sqldb}:${DB_PORT:=3306}/${DB_BASE:=kimai}"
+    elif [ "$DB_TYPE" == "sqlite" ]; then
+      export DATABASE_URL="${DB_TYPE}://${DB_BASE:=/%kernel.project_dir%/var/data/kimai.sqlite}"
```
Comment (kevinpapst) [inline startup.sh]: Is the slash in frontend of `/%kernel.project_dir%/` correct?
Or should it be 
```
      export DATABASE_URL="${DB_TYPE}://${DB_BASE:=%kernel.project_dir%/var/data/kimai.sqlite}"
```
?

**Review by kevinpapst — CHANGES_REQUESTED** [review]

File: startup.sh (hunk 2 above)
Comment (tobybatch) [reply startup.sh]: Yes (I think).  I've tweaked the docs too, but it allows for either relative to absolute path to be passed to the startup.

Comment (kevinpapst) [reply startup.sh]: AFAIk `%kernel.project_dir%` is always an absolute path to the kimai directory, so here `/opt/kimai`. If a user wants to move it somewhere else, one could use `DB_BASE=/tmp/kimai.sqlite`. The slash in front of `%kernel.project_dir%` might not be a problem as `/opt/kimai` and `//opt/kimai` will be handled the same, but is unnecessary.

File: docs/runtime-args.md [hunk 3]
```diff
@@ -21,3 +21,21 @@ If set then these values will try and create a new admin user.
...
+
+ * ```DB_USER``` defaults to ```kimai```
+ * ```DB_PASS``` defaults to ```kimai```
+ * ```DB_HOST``` defaults to ```sqldb```
+ * ```DB_PORT``` defaults to ```3306```
+ * ```DB_BASE``` defaults to ```kimai```
+
+If the ```DB_TYPE``` is sqlite:
+
+ * ```DB_BASE``` defaults to ```%kernel.project_dir%/var/data/kimai.sqlite```, N.B. This should be the path realtive to ```/opt/kimai```, prefix this with a ```/``` to path it from the root of the image.
```
Comment (kevinpapst) [inline docs/runtime-args.md]: realtive <=> relative

File: startup.sh [hunk 4]
```diff
@@ -4,13 +4,24 @@ echo $KIMAI
...
+    else
+      echo "Unkown database type, cannot proceed. [$DB_TYPE]"
```
Comment (kevinpapst) [inline startup.sh]: Unkown <=> Unknown

File: startup.sh (hunk 2 above)
Comment (tobybatch) [reply startup.sh]: I'll snip it anyway, hang on

**Review by kevinpapst — APPROVED** [review]

**Pull request merged**